runtime/cudaq/platform/pasqal/PasqalServerHelper.cpp

```diff
@@ -52,50 +55,60 @@ RestHeaders PasqalServerHelper::getHeaders() {
 
 ServerJobPayload
 PasqalServerHelper::createJob(std::vector<KernelExecution> &circuitCodes) {
-  ServerJobPayload ret;
-  std::vector<ServerMessage> &tasks = std::get<2>(ret);
+  std::vector<ServerMessage> tasks;
+
   for (auto &circuitCode : circuitCodes) {
-    // Construct the job message
-    ServerMessage taskRequest;
+    ServerMessage message;
+    message["name"] = circuitCode.name;
+    message["device"] = backendConfig.at("machine");
+    message["shots"] = shots;
 
-    taskRequest["name"] = circuitCode.name;
-    taskRequest["device"] = backendConfig.at("machine");
     auto action = nlohmann::json::parse(circuitCode.code);
-    taskRequest["action"] = action.dump();
-    taskRequest["shots"] = shots;
+    message["action"] = action.dump();
 
-    tasks.push_back(taskRequest);
+    tasks.push_back(message);
   }
 
   cudaq::info("Created job payload for Pasqal, targeting device {}",
               backendConfig.at("machine"));
   
   // Return a tuple containing the job path, headers, and the job message
-  return std::make_tuple(baseUrl + "/v1/cuda-q/job", getHeaders(), tasks);
+  return std::make_tuple(baseUrl + apiPath + "/cudaq/job", getHeaders(), tasks);
 }
 
 std::string PasqalServerHelper::extractJobId(ServerMessage &postResponse) {
-    return baseUrl + "";
+    return postResponse["id"].get<std::string>();
 }
 
 std::string PasqalServerHelper::constructGetJobPath(std::string &jobId) {
-  return baseUrl + "";
+  return baseUrl + apiPath + "/jobs/" + jobId + "/results_link";
```

Comment:
I think the best thing here will be that we re-use the existing API, so
the full url would be sth like
```
https://apis.pasqal.com/api/v1/batches/<jobid>/results
```
as detailed here:
https://docs.pasqal.cloud/cloud/api/core/operations/get_results_api_v1_batches__batch__id__results_get/